parseAndPopulate/integrity.py

```diff
@@ -86,27 +86,26 @@ def __init__(self):
 def check_revision(parsed_module: Statement) -> bool:
     try:
         revision = parsed_module.search('revision')[0].arg
-    except:
+    except (IndexError, AttributeError):
         return False
     revision_parts = [int(i) for i in revision.split('-')]
     try:
         date(*revision_parts)
-    except:
+    except ValueError:
         if revision_parts[1:] == [2, 29]:
             revision_parts[2] = 28
             try:
                 date(*revision_parts)
-            except:
+            except ValueError:
                 return False
-        else:
-            return False
+        return False
```

Comment:
This would return False, even if we can fix the date.

Reply:
I just removed the ```else``` statement, that doesn't really change anything, so ```if revision_parts[1:] != [2, 29]:``` then ``False``` will be returned, we just can skip the ```else``` statement here

Reply:
Yes, but now `False` will also be returned even if `revision_parts[1:] == [2, 29]` is `True`.

Reply:
Yes, but now `False` will also be returned even if `revision_parts[1:] == [2, 29]` is `True`.

Reply:
Oh, I see, thanks, fixed